crates/uv/tests/common/mod.rs

```diff
@@ -862,6 +863,28 @@ pub fn copy_dir_ignore(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> anyhow::
     Ok(())
 }
 
+/// Create a stub package `name` in `dir` with the given `pyproject.toml` body.
+pub fn make_project(dir: &Path, name: &str, body: &str) -> anyhow::Result<()> {
+    let pyproject_toml = formatdoc! {r#"
+        [project]
+        name = "{name}"
+        version = "0.1.0"
+        description = "Test package for direct URLs in branches"
+        requires-python = ">=3.11,<3.13"
+        {body}
+
+        [build-system]
+        requires = ["flit_core>=3.8,<4"]
+        build-backend = "flit_core.buildapi"
```

Comment:
Random question, but how we specify a build system here? Should I be doing this in my own ad hoc experiments?

Reply:
If you specify nothing, we default to setuptools. Setuptools works, but has some quirks like creating an `.egg-info` directory and not checking if there's actual code for your package, so i normally go with hatch for my experiments:

```
[build-system]
requires = ["hatchling"]
build-backend = "hatchling.build"
```

Flit is another modern, more minimalistic build backend.